In [1]:
import sys
sys.path.append('../src')

from pytorch_lightning import Trainer
from model_gnn import GNN_Module
from min_max_graph_data import MinMaxGraphModule

In [2]:
data = MinMaxGraphModule()
model = GNN_Module()
trainer = Trainer(check_val_every_n_epoch=5, gpus=1, max_epochs=150)

GPU available: True, used: True
TPU available: False, using: 0 TPU cores
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


In [3]:
trainer.fit(model, datamodule=data)


  | Name         | Type             | Params
--------------------------------------------------
0 | model        | GNN_Model        | 951 K 
1 | loss         | CrossEntropyLoss | 0     
2 | val_accuracy | Accuracy         | 0     


Validation sanity check: 0it [00:00, ?it/s]

/home/kushaj/miniconda3/envs/dl/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:45: UserWarning: The dataloader, val dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 24 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  warnings.warn(*args, **kwargs)


Epoch 0:   0%|          | 2/400 [00:00<00:34, 11.50it/s, loss=358.165, v_num=15]

/home/kushaj/miniconda3/envs/dl/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:45: UserWarning: The dataloader, train dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 24 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  warnings.warn(*args, **kwargs)


Epoch 4: 100%|██████████| 390/390 [00:23<00:00, 16.47it/s, loss=1.981, v_num=15] 
Validating: 0it [00:00, ?it/s]
Validating: 1it [00:00,  9.31it/s]
Epoch 4: : 392it [00:23, 16.41it/s, loss=1.981, v_num=15]                       
Validating: 3it [00:00,  9.29it/s]
Epoch 4: : 394it [00:24, 16.34it/s, loss=1.981, v_num=15]
Validating: 5it [00:00,  9.25it/s]
Epoch 4: : 396it [00:24, 16.28it/s, loss=1.981, v_num=15]
Validating: 7it [00:00,  9.28it/s]
Epoch 4: : 398it [00:24, 16.22it/s, loss=1.981, v_num=15]
Epoch 4: : 400it [00:24, 16.17it/s, loss=1.981, v_num=15, val_loss=2.28, acc=0.131]
Epoch 9: 100%|██████████| 390/390 [00:23<00:00, 16.74it/s, loss=1.736, v_num=15, val_loss=2.28, acc=0.131]
Validating: 0it [00:00, ?it/s]
Validating: 1it [00:00,  9.51it/s]
Epoch 9: : 392it [00:23, 16.67it/s, loss=1.736, v_num=15, val_loss=2.28, acc=0.131]                       
Validating: 3it [00:00,  9.45it/s]
Epoch 9: : 394it [00:23, 16.61it/s, loss=1.736, v_num=15, val_loss=2.28, acc=0.131]
Validatin

Epoch 59: : 392it [00:22, 17.65it/s, loss=1.327, v_num=15, val_loss=0.146, acc=0.971]                       
Validating: 3it [00:00, 10.07it/s]
Epoch 59: : 394it [00:22, 17.58it/s, loss=1.327, v_num=15, val_loss=0.146, acc=0.971]
Epoch 59: : 396it [00:22, 17.51it/s, loss=1.327, v_num=15, val_loss=0.146, acc=0.971]
Validating: 7it [00:00, 10.02it/s]
Epoch 59: : 398it [00:22, 17.44it/s, loss=1.327, v_num=15, val_loss=0.146, acc=0.971]
Epoch 59: : 400it [00:22, 17.40it/s, loss=1.327, v_num=15, val_loss=1.11, acc=0.423] 
Epoch 64: 100%|██████████| 390/390 [00:22<00:00, 17.03it/s, loss=0.341, v_num=15, val_loss=1.11, acc=0.423] 
Validating: 0it [00:00, ?it/s]
Validating: 1it [00:00,  9.83it/s]
Epoch 64: : 392it [00:23, 16.96it/s, loss=0.341, v_num=15, val_loss=1.11, acc=0.423]                       
Validating: 3it [00:00,  9.80it/s]
Epoch 64: : 394it [00:23, 16.90it/s, loss=0.341, v_num=15, val_loss=1.11, acc=0.423]
Validating: 5it [00:00,  9.83it/s]
Epoch 64: : 396it [00:23, 16.84it/s, lo

Validating: 3it [00:00,  9.48it/s]
Epoch 119: : 394it [00:22, 17.40it/s, loss=0.086, v_num=15, val_loss=0.204, acc=0.926]
Validating: 5it [00:00,  9.60it/s]
Epoch 119: : 396it [00:22, 17.34it/s, loss=0.086, v_num=15, val_loss=0.204, acc=0.926]
Validating: 7it [00:00,  9.45it/s]
Epoch 119: : 398it [00:23, 17.26it/s, loss=0.086, v_num=15, val_loss=0.204, acc=0.926]
Epoch 119: : 400it [00:23, 17.20it/s, loss=0.086, v_num=15, val_loss=0.0878, acc=0.983]
Epoch 124: 100%|██████████| 390/390 [00:22<00:00, 17.73it/s, loss=0.120, v_num=15, val_loss=0.0878, acc=0.983] 
Validating: 0it [00:00, ?it/s]
Epoch 124: : 392it [00:22, 17.66it/s, loss=0.120, v_num=15, val_loss=0.0878, acc=0.983]                       
Validating: 3it [00:00, 10.08it/s]
Epoch 124: : 394it [00:22, 17.59it/s, loss=0.120, v_num=15, val_loss=0.0878, acc=0.983]
Validating: 5it [00:00, 10.00it/s]
Epoch 124: : 396it [00:22, 17.52it/s, loss=0.120, v_num=15, val_loss=0.0878, acc=0.983]
Validating: 7it [00:00,  9.93it/s]
Epoch 124: 

1

The best model achieved 98% accuracy. This is without hyperparameter optimization but we can still observe from this that GNN can solve this problem. The fluctutations in validation accuracy are probably due to similiar distribution of validation data i.e. the random sampling of data might have sampled data in the same region.